### PR TITLE
Entity-aware content layouts

### PR DESCRIPTION
**Pull request by hypeJunction**
Alternative to #7870

### CODE REVIEW

**Comment by hypeJunction** [comment]
I basically need an entity in the title menu. On a client site I need to get rid of the entity menu, and combine it with the title menu.

File: mod/groups/views/default/resources/groups/profile.php [hunk 1]
```diff
@@ -50,7 +50,9 @@
 	'content' => $content,
 	'sidebar' => $sidebar,
 	'title' => $group->name,
+	'entity' => $group,
+	'filter' => '',
 );
-$body = elgg_view_layout('one_sidebar', $params);
+$body = elgg_view_layout('content', $params);
```
Comment (mrclay) [inline mod/groups/views/default/resources/groups/profile.php]: We definitely can't switch a layout in a minor release.

Comment (hypeJunction) [inline mod/groups/views/default/resources/groups/profile.php]: Right, we can keep it as one_sidebar, and upgrade for 3.0

Comment (hypeJunction) [inline mod/groups/views/default/resources/groups/profile.php]: Btw, can we setup 3.x branch already?

**Comment by mrclay** [comment]
If a 2.0 plugin overrides one of these resource views, the entity won't be propagated to where it's expected, so I don't see how to make this BC.

Seems like a good idea though.

**Comment by hypeJunction** [comment]
This is a feature, which we can add, but not build upon in core until 3.x

**Comment by hypeJunction** [comment]
Or we can just wait till 3.x. I can just update the resource views in my project. Can we already make breaking change PRs to master for 3.x?

**Comment by mrclay** [comment]
Yes, put it on master. We'll break off 3.0 and 3.x later. This will need a breaking change notice and upgrade docs explaining that the content layout and page/layouts/elements/header views now require "entity" when displaying a single entity.

We probably want to wait for 2.1.0 before merging stuff into master to avoid conflict hell.

**Comment by hypeJunction** [comment]
Ok, will do. As a next step, I would also like to add entity to page shell vars, and make entity available to head hooks, so you can use entity to generate html meta tags.

**Comment by mrclay** [comment]
SGTM. The limitation for 2.x being core can't rely on new values coming from views, unless it's for new functionality (that would be harmless if left out). Meta tags are a good example.

**Comment by hypeJunction** [comment]
Open graph optimists would argue that meta tags are more important than the whole of html content ;)

**Comment by hypeJunction** [comment]
I would like to revive and expand this. I think it's by far the only transparent way to deal to deal with entity data in HTML shells without sniffing the URLs.

See https://github.com/Elgg/community_plugins/pull/139 for a use case.

Entity menu would be good to have, but not essential to establish a convention.

**Pull request closed without merging**